UITests/UITests.Tests.Shared/Tests.cs

```diff
@@ -113,5 +115,33 @@ public void SimpleLaunchTest()
 
             Verify.AreEqual("Clicked", textBlock.GetText());
         }
+
+        [TestMethod]
+        public void TestTextBoxMaskBinding_Property()
+        {
+            OpenTest("TextBox Mask");
+
+            var initialValue = FindElement.ById<TextBlock>("InitialValueTextBlock").GetText();
+            var textBox = FindElement.ById<Edit>("TextBox");
+
+            Verify.AreEqual(initialValue, textBox.GetText());
+
+            var changeButton = FindElement.ById<Button>("ChangeButton");
+
+            changeButton.Click();
+            Wait.ForIdle();
+
+            var newValue = FindElement.ById<TextBlock>("NewValueTextBlock").GetText();
+
+            Verify.AreEqual(newValue, textBox.GetText());
+        }
+
+        private static void OpenTest(string name)
```

Comment:
Yeah, so this method should be integrated into the test-harness itself and know how to communicate with the MainPage in order to switch the page in the frame. We don't want to have to rely on adding more custom buttons and things to the main test page in order to do this.

As mentioned in the other PR I think the use of a custom Attribute could be handy if we could make that work. i.e.

```cs
    [TestMethod]
    [TestXamlFile("TextBoxMaskTestPage")]
    public void TestTextBoxMaskBinding_Property()
    {
```

The test harness before each test would read the attribute on it to grab the type name of the page to load, then we'd have to send the test app a message to load that page, and then execute the test.

Not sure if we would need an intermediary UI piece, use the clipboard, or go as heavy as an AppService to do this. Looks like in WinUI they just manipulate the UI similar to how you did it here, just more generally with a command: https://github.com/microsoft/microsoft-ui-xaml/blob/master/test/TestAppUtils/NavigateToTestCommand.cs#L12

And they use their [TestSetupHelper](https://github.com/microsoft/microsoft-ui-xaml/blob/master/test/testinfra/MUXTestInfra/Infra/TestHelpers.cs#L42) to manually control the page at the start of each test: https://github.com/microsoft/microsoft-ui-xaml/blob/master/dev/TabView/InteractionTests/TabViewTests.cs#L49

@azchohfi thoughts? Should we just copy the setup WinUI has or should we try and improve upon this somehow with an attribute on the test method itself to control this?

I'm guessing it is easier if the idea is to better introp to WinUI that we more closely follow their patterns? @ranjeshj any preferences on if we experiment or try to more closely align? Has the current WinUI system been working well or have you been looking to changing it in the future?

Reply:
I guess I left out that part of why I want to try and simplify these connections is to make it as straight-forward and easy for developers to add integration tests so that we can ensure adding them isn't a big burden to our community.

I think if we could introduce a new attribute based system here, it'd give developers the flexibility to re-use a page easily or setup multiple pages for different scenarios, eh?

It also makes each test easier to maintain and understand as all the test harness initialization is abstracted to the attribute.

Reply:
Basically we'd set something up like this to grab the attribute for the base of our test system:

```cs
public TestContext TestContext { get; set; }

[TestInitialize]
public void TestInitialize()
{
    Debug.WriteLine("Running Test: " + TestContext.TestName);

    var currentlyRunningMethod = GetType().GetMethod(TestContext.TestName);
    var descriptionAttributes = currentlyRunningMethod.GetCustomAttributes(typeof(DescriptionAttribute), true) as System.Collections.Generic.IEnumerable<DescriptionAttribute>;
    var firstOne = descriptionAttributes.FirstOrDefault();
    if (firstOne != null)
    {
        Debug.WriteLine("Description: " + firstOne.Description);
    }
}
```

This is just grabbing a description attribute, but would work the same for our custom attribute and use that info to load the appropriate page in the Test App via test control commands as done currently or some other channel (AppService?).

In either case, I think we should have the test app load the requested page on demand vs. trying to determine all the pages ahead of time and load them at the start of the app. I think this would help with the test app starting time? Which would help if you're picking specific tests to run from VS, right?

Thoughts?